### ISSUE TITLE
[Feature Request] Aseprite file format support.

### ISSUE THREAD

**Issue by DeLoR1AnEc** [TagStudioDev/TagStudio#171]
The title says for itself.

**Comment by mak448a** [comment]
Quick note here: you should look at libresprite (fork of aseprite before license change) for the file format, since aseprite isn't open source anymore.

**Comment by DeLoR1AnEc** [comment]
Wait really? What else did I missed...

**Comment by mak448a** [comment]
> Wait really? What else did I missed...
> 
> Edit: I checked this and it seems that the source code is still available and there were no announcements on their official site.

Yes, the source is available, but you can't modify, change, or use any code from it.

**Comment by Cool-Game-Gev** [comment]
I would also like this to be added, but I don’t know how difficult it would be.

**Comment by goldstargloww** [comment]
i would also like this! ^^

**Comment by TrigamDev** [comment]
There's several ways of going about this:

## Use the official [Aseprite CLI](https://www.aseprite.org/docs/cli/)

### Pros
- Simplest to do.
- Most stable and reliable, since it uses Aseprite directly.
- We don't have to create or maintain the renderer.

### Cons
- Requires the user to have Aseprite installed.
  - This is likely for people who have and are working with Aseprite files, but not guaranteed.
  - The user can't be prompted to just install Aseprite, as Aseprite isn't free (built versions, anyway).

## Use a third-party library

- Some libraries to take a look at:
  - https://docs.rs/asefile/latest/asefile/
  - https://docs.rs/aseprite-io/latest/aseprite/
  - https://docs.rs/aseprite-loader/latest/aseprite_loader/

### Pros
- Doesn't require the user to install Aseprite.
- We don't have to create or maintain the renderer.

### Cons
- Since it's not the official Aseprite CLI, there's no guarantee that it's 100% accurate.
- Due to it being an external library, it's more difficult to resolve any issues that may be present.

## Create an in-house renderer

### Pros
- Doesn't require the user to install Aseprite.
- It's easier to make changes and fixes as needed.

### Cons
- Since it's not the official Aseprite CLI, there's no guarantee that it's 100% accurate.
- We bear the burden of needing to create and maintain the renderer.
- We may be more likely to make mistakes, as the focus isn't solely on the Aseprite thumbnail rendering.
